ray_lightning/ray_ddp.py

```diff
@@ -514,8 +526,9 @@ def node_rank(self) -> int:
     def root_device(self):
         if self.use_gpu and torch.cuda.is_available():
             if self._is_remote:
-                # Adjust for if there are multiple GPUs per worker.
-                device_id = self.local_rank * self.num_gpus_per_worker
+                # Adjust to support multiple GPUs per worker or fractional
+                # GPUs per worker.
+                device_id = int(self.local_rank * self.num_gpus_per_worker)
```

Comment:
nit: This might lead to strange behavior when 1 can't be evenly divided by `num_gpus_per_worker`.

E.g. when `num_gpus_per_worker=0.4`, `device_id=0` for `local_ranks` 0, 1, 2.

Reply:
Yeah this is a good point- switched to just using `ray.get_gpu_ids()` and added a test for this.